### DISCUSSION TITLE
BMS registers available over a LAN connection

### DISCUSSION THREAD

**Discussion by canton7** [TonyM1958/HA-FoxESS-Modbus#9] [Q&A] [answered]
Hi,

I was just going through the latest spreadsheet, and noticed that you had the following registers down as being available over a direct LAN connection:

* BMS Max Charge Current
* BMS Max Discharge Current

The conventional wisdom is that all BMS registers aren't available over a direct LAN connection. Have you actually confirmed that these ones are?

Thanks!

**Comment by TonyM1958** [comment, marked as answer]
Yes, on H1, these holding registers are present. They not be exactly what you think though?

BMS Temperature is a rename of Battery Temperature (keeps unique ID) as this is a better description of what it is.

BMS Max Charge / Discharge Current are feedback from the BMS to the inverter - you can see these as requests from the BMS to the inverter.

So, when the BMS wants charge, it sets BMS Max Charge Current (typically to 30A) and when the battery is full, it sets this to zero. The inverter responds by sending the smaller of the requested charge current and the Max Charge Current in the inverter parameters. Using this the BMS can throttle the charge current, if it wants to.

Conversely, the BMS Max Discharge current does the same but during discharge. When the battery has a full charge, the BMS sends 50A available to the inverter. The inverter draws the power it needs, up to the smaller of this value and the inverter Max Discharge Current parameter. When the batteries are getting empty, the BMS reduces the Max Current until it reaches zero, telling the inverter it can no longer provide power.

For the most part, I don't actively monitor these values but pulling them up and tracking the history against SoC can be useful if you think there is a problem with your batteries.

BTW - presume you noticed that the inverter settings password is 40013 and is read/write as a decimal number... just in case anyone does not know their password or needs to reset it? Just need to work on the morale dilema of whether you should expose this if the installer has set it to prevent end user access...

**Reply by canton7** [reply]
Thanks, looks like I can map the charge/discharge currents then!

**Reply by canton7** [reply]
I hadn't spotted the inverter password there... I'm tempted not to expose that - it's just going to be clutter almost all of the time!

**Reply by TonyM1958** [reply]
Yes, interesting one. Will niggle a bit every time someone posts asking if you know what the password is ;-). You can just write it as well, so it's equally possible to have a 'reset password button' that sets it to zero.

**Reply by canton7** [reply]
Yeah, has anyone written that and had a password which wasn't 0000? At a pinch we can always tell people to use the write register service

**Comment by TonyM1958** [comment]
A question for you... I've been looking at enabling / disabling force charge from grid using an automation to write to 41001 but this always seems to come back with an error. Don't know if you've looked at this, as you probably write directly. Is there a trick to getting this to work using the HA modbus write services?

**Reply by canton7** [reply]
You need to set the whole batch of 6 charge period sensors together.

**Reply by TonyM1958** [reply]
Great thanks - presume that is write multiple registers, no writing coils or anything for the binary sensors? (do they even need to be binary, just read a word as 0/1 as on/off in HA seems pretty clunky)

**Reply by canton7** [reply]
Yep, just a write multiple registers which spans all 6.

If those ones were coils you wouldn't be able to read/write them in one go along with other sensors, so what they've done does make sense IMO.

**Reply by TonyM1958** [reply]
Thanks. Don't get too excited by the BMS current registers,  the only values I've ever seen are 30A and 0A for charge and 50A and 0A for discharge, so pretty much charge / discharge on or off...

**Reply by canton7** [reply]
Good shout, I've changed them to be disabeld by default

**Reply by TonyM1958** [reply]
> Yep, just a write multiple registers which spans all 6.

Thanks - all working now following your good advice. Also swapped the spread sheet and my integration to use standard sensor instead of binary sensor as that simplifies things.
